xbmc/windowing/osx/WinSystemOSX.mm

```diff
@@ -1151,16 +1210,24 @@ static void DisplayReconfigured(CGDirectDisplayID display,
   const NSUInteger dispIdx =
       GetDisplayIndex(CServiceBroker::GetSettingsComponent()->GetSettings()->GetString(
           CSettings::SETTING_VIDEOSCREEN_MONITOR));
-  NSScreen* screen = nil;
-  if (dispIdx < NSScreen.screens.count)
+  NSScreen* currentScreen = nil;
+  NSScreen* targetScreen = nil;
```

Comment:
just FYI (not necessary to change): under ARC (which we use) all objc objects are nil-initialised implicitly